### PR TITLE
Switch to the REST-based permissions API

### PR DESCRIPTION
**Pull request by tgoyne**
Fixes https://github.com/realm/realm-cocoa-private/issues/433

### CODE REVIEW

**Comment by tgoyne** [comment]
The Cocoapods CI failures are a CI issue.

File: Realm/RLMSyncUser.mm [hunk 1]
```diff
@@ -391,130 +385,197 @@ + (void)confirmEmailForAuthServer:(NSURL *)serverURL
...
-    verifyInRunLoop();
-    if (!_user || _user->state() == SyncUser::State::Error) {
-        callback(make_permission_error_change(@"Permissions cannot be revoked using an invalid user."));
-        return;
-    }
-    Permissions::delete_permission(_user,
-                                   [permission rawPermission],
-                                   RLMWrapPermissionStatusCallback(callback),
-                                   *_configMaker);
+static id ISO8601Formatter() {
```
Comment (jsflax) [inline Realm/RLMSyncUser.mm]: [q] Why not return `NSDateFormatter`? Also, could this be in a utility class instead of on the SyncUser?

Comment (tgoyne) [reply Realm/RLMSyncUser.mm]: It was conditionally returning NSISO8601DateFormatter if running on an OS version which supports it, but that turned out not to work and I forgot to change the return type. I guess I might as well toss it in RLMUtil.

**Comment by tgoyne** [comment]
@fealebenpae @nirinchev This could use a review from one of you two to verify that the calls to the server look right to you.

**Pull request merged**